apps/server/src/shared/domain/entity/task.entity.spec.ts

```diff
@@ -28,6 +37,495 @@ describe('Task Entity', () => {
 		});
 	});
 
+	describe('getSubmittedUserIds', () => {
+		it('should use save private getSubmissionItems methode to fetch submissions', () => {
+			const task = taskFactory.build();
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any
+			const spy = jest.spyOn(Task.prototype as any, 'getSubmissionsItems');
+
+			task.getSubmittedUserIds();
+
+			expect(spy).toHaveBeenCalled();
+		});
+
+		it.todo('should validate if data are loaded');
+
+		describe('when submissions are loaded', () => {
+			it('should return a list of submitted userIds', () => {
+				const student = userFactory.build();
+				student.id = '0123456789ab';
+
+				const submission = submissionFactory.build({ student });
+				const task = taskFactory.build();
```

Comment:
could be maybe moved before building the submission and passed to the submission factory as well:
 ```typescript
 const task = taskFactory.build();
 submissionFactory.build({ task, student });
 ```

Reply:
yes true